Lib/concurrent/futures/thread.py

```diff
@@ -148,4 +150,5 @@ def shutdown(self, wait=True):
         if wait:
             for t in self._threads:
                 t.join()
+        self._threads.clear()
```

Comment:
Also, there is no race condition here. Besides the Thread object doesn't seem to hold onto any important data, so it shouldn't be a problem if isn't collected immediately.